utils/ocm.py

```diff
@@ -491,6 +492,57 @@ def create_kafka_cluster(self, data):
         api = '/api/managed-services-api/v1/kafkas?async=true'
         self._post(api, data)
 
+    def _init_addons(self):
+        """Returns a list of Addons """
+        api = '/api/clusters_mgmt/v1/addons'
+        self.addons = self._get_json(api).get('items')
+
+    def get_addon(self, name):
+        for addon in self.addons:
+            resource_name = addon['resource_name']
+            if name == resource_name:
+                return addon
+        return None
+
+    def get_cluster_addons(self, cluster):
+        """Returns a list of Addons installed on a cluster
+
+        :param cluster: cluster name
+
+        :type cluster: string
+        """
+        results = []
+        cluster_id = self.cluster_ids.get(cluster)
+        if not cluster_id:
+            return results
+        api = \
+            f'/api/clusters_mgmt/v1/clusters/{cluster_id}/addons'
+        items = self._get_json(api).get('items')
+        if not items:
+            return results
+
+        for item in items:
+            desired_keys = ['id']
+            result = {k: v for k, v in item.items() if k in desired_keys}
+            results.append(result)
+
+        return results
+
+    def install_addon(self, cluster, spec):
+        """ Installs an addon on a cluster
+
+        :param cluster: cluster name
+        :param spec: required information for installation
```

Comment:
hard to keep track of what that would be (the required information).  I see you're using the diff, but that is built, then some keys are popped... as you're at it, could you please document here what is expected in the spec?

Reply:
+1 I wasn't able to find the spec definition. Might be okay to add a link to the api.openshift.com docs here as the spec is evolving?

Reply:
updated!